pkg/registry/migrate.go

```diff
@@ -0,0 +1,150 @@
+package registry
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	clusterv1beta1 "github.com/replicatedhq/embedded-cluster-kinds/apis/v1beta1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const RegistryDataMigrationCompleteSecretName = "registry-data-migration-complete"
+const registryDataMigrationJobName = "registry-data-migration"
+
+const RegistryMigrationStatusConditionType = "RegistryMigrationStatus"
+const RegistryMigrationServiceAccountName = "registry-data-migration-serviceaccount"
+
+// MigrateRegistryData should be called when transitioning from non-HA to HA airgapped installations
+// this function scales down the registry deployment to 0 replicas, then creates a job that will migrate the data before
```

Comment:
Might wanna update the comment here as it seems this function doesn't scale the registry anymore?

Reply:
yeah will do